builtin/providers/aws/resource_aws_kinesis_analytics.go

```diff
@@ -0,0 +1,541 @@
+				ResourceARN: aws.String(input["arn"].(string)),
+				RoleARN:     aws.String(input["role_arn"].(string)),
+			},
+		}
+
+		if *i.InputSchema.RecordFormat.RecordFormatType == "JSON" {
+			i.InputSchema.RecordFormat.MappingParameters = &kinesisanalytics.MappingParameters{
+				JSONMappingParameters: &kinesisanalytics.JSONMappingParameters{
+					RecordRowPath: aws.String(input["record_row_path"].(string)),
+				},
+			}
+		} else if *i.InputSchema.RecordFormat.RecordFormatType == "CSV" {
+			i.InputSchema.RecordFormat.MappingParameters = &kinesisanalytics.MappingParameters{
+				CSVMappingParameters: &kinesisanalytics.CSVMappingParameters{
+					RecordColumnDelimiter: aws.String(input["record_column_delimiter"].(string)),
+					RecordRowDelimiter:    aws.String(input["record_row_delimiter"].(string)),
+				},
+			}
+		} else {
+			return nil, fmt.Errorf("format must be either 'JSON or CSV'. you gave an unsupported record format type: %s", *i.InputSchema.RecordFormat.RecordFormatType)
+		}
+
+		inputStreams = append(inputStreams, i)
+	}
+
+	return inputStreams, nil
+}
+
+func resourceAwsKinesisAnalyticsCreate(d *schema.ResourceData, meta interface{}) error {
+
+	conn := meta.(*AWSClient).kinesisanalyticsconn
+
+	name := d.Get("name").(string)
+	appDesc := d.Get("application_description").(string)
+	appCode := d.Get("application_code").(string)
+	inputs := d.Get("inputs").(*schema.Set).List()
+	outputs := d.Get("outputs").(*schema.Set).List()
+
+	inputStreams, err := createInputList(inputs)
+
+	if err != nil {
+		return err
+	}
+
+	outputStreams, err := createOutputList(outputs)
+
+	if err != nil {
+		return err
+	}
+
+	//fmt.Printf("KA.INPUTS:  %T\n\n", inputs)
+	//fmt.Printf("KA.INPUTS:  %+v\n\n", inputs)
+	//
+	//element := inputs[0].(map[string]interface{})
+	//fmt.Printf("element:  %T\n\n", element)
+	//fmt.Printf("property access:  %+v\n\n", element["name"].(string))
+
+	createOpts := &kinesisanalytics.CreateApplicationInput{
+		ApplicationName:        aws.String(name),
+		ApplicationDescription: aws.String(appDesc),
+		ApplicationCode:        aws.String(appCode),
+		Inputs:                 inputStreams,
+		Outputs:                outputStreams,
+	}
+
+	_, err = conn.CreateApplication(createOpts)
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			return fmt.Errorf("[WARN] Error creating Kinesis Analytics Application: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
+		}
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"CREATING"},
+		Target:     []string{"READY"},
+		Refresh:    applicationStateRefreshFunc(conn, name),
+		Timeout:    3 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	state, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for Kinesis Analytics Application (%s) to become active: %s",
+			name, err)
+	}
+
+	s := state.(*KinesisAnalyticsState)
+	d.SetId(s.arn)
```

Comment:
If ARN is a pointer, we need to dereference here - SetId doesn't dereference for us